### ISSUE TITLE
Scrolling over diff/log jumps around

### ISSUE THREAD

**Issue by remram44** [tibirna/qgit#105]
I think this is intended, but it's really annoying.

When scrolling down past the end of the log, the view changes to diff. When scrolling down past the end of the diff, the view jumps UP to the NEXT commit. Even worse, scrolling back up goes again to the NEXT commit.

It's a bit annoying to be scrolling down a diff and to suddenly have the view jump to a different commit.

There are additional glitches when switching from log view to diff view that way (scrolling down), it won't be possible to scroll through the diff (have to stop scrolling my mousewheel for a few seconds before scrolling works again).

I would appreciate an option to disable this whole thing, though I suspect it could be removed altogether.

**Comment by martenjj** [comment]
Don't know whether this is intended behaviour, but it seems that you can stop scrolling switching from log to diff, and scrolling down from diff going to a different commit, by turning off "Show smart labels" in Settings - Browse.

**Comment by remram44** [comment]
Well that's embarrassing... :sweat_smile:

I had noticed the option for tabbed view, but not that particular checkbox. Though it could probably be made more explicit about what it does (it's not clear what "smart labels" are, and the scrolling behavior is not mentioned), unchecking it has fixed the problem for me.

Thank you!

**Comment by WickedSmoke** [comment]
I agree that this behavior is bothersome and also had no idea that it could be disabled.   Yes, at a minimum the name and/or tool tip should mention scrolling.

This seems like a deeper design problem.  I'd really like the Log to be visible along side the Diff and not hide it.

**Comment by WickedSmoke** [comment]
After thinking on it a bit, simply unifying the Log & Diff tabs would be my preferred layout.  Similar to how "git show" or gitweb display commits.  This would streamline the interface (no tabs or smart labels needed) and the scrolling option could then be called something like "Allow mouse wheel to change commits".

Some option combinations seem useless.  If both "Show tabbed revision msg / diff pane" & "Show smart labels" are disabled then the current view (with a now hidden tab) is locked and there is no way to see the other Log/Diff content.
